src/formatDate.js

```diff
@@ -8,7 +8,36 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromFormatSeparator = fromFormat[3];
+  const toFormatSeparator = toFormat[3];
+  const dateFromParts = date.split(fromFormatSeparator);
+  const formatingFrom = {};
+  const dateToParts = [];
+
+  formatingFrom[fromFormat[0]] = dateFromParts[0];
+  formatingFrom[fromFormat[1]] = dateFromParts[1];
+  formatingFrom[fromFormat[2]] = dateFromParts[2];
+
+  for (let i = 0; i < 3; i++) {
+    for (const key in formatingFrom) {
+      if (key.includes(toFormat[i][0])) {
+        const needLength = toFormat[i].length;
+        const isLength = formatingFrom[key].length;
+        let newPartDate = formatingFrom[key];
+
+        if (isLength !== needLength && key.includes('Y')) {
+          if (needLength === 2) {
+            newPartDate = formatingFrom[key].slice(2, 4);
+          } else {
+            newPartDate = (newPartDate < 30 ? '20' : '19') + newPartDate;
+          }
+        }
```

Comment:
The condition `newPartDate < 30` is not sufficient to determine the century. It only works if `newPartDate` is a 2-digit year. If `newPartDate` is already a 4-digit year, this logic will incorrectly prepend '20' or '19' to it. You should first check if you're working with a 2-digit year before applying this logic.